### PR TITLE
Fix Open Online Link in DropDown 

### LINKED ISSUE #808: Add an 'Open online' button in the simulator menu
**Describe the solution you'd like**
Currently, the project menu in the simulator shows options for managing a project and opening an offline project, but it doesn't have a button for opening an online project.

Screenshot 2020-01-05 at 12 17 30

Although a user's projects can be viewed and edited on their projects page, there should be an `Open Online` button which takes the user straight to this page

Edit:
let the option be **view project page** and it should open the corresponding project page and not the user page. 

**Are you working on this? (Yes/No)**
I can if needed. Suggested task for GCI.

### PR DESCRIPTION
**Pull request by Shivansh2407**
Fixes #808

#### Describe the changes you have made in this pr -
Added Open Online link to the Dropdown.
@JoshHeng I need your help in making the data.js . I wanted to know whether I can create a function in that JS file with content like this window.location = "users/#{@user.id}" ?

### CODE REVIEW

**Comment by JoshHeng** [comment]
> @JoshHeng I need your help in making the data.js . I wanted to know whether I can create a function in that JS file with content like this window.location = "users/#{@user.id}" ?

You should be able to. Follow the format of the other links, and test it to see whether it works!

**Comment by thiefie** [comment]
Till the time are working on this you could add a [WIP] in the PR title.

**Comment by Shivansh2407** [comment]
Will Fix the Code Climate Issues Later. I would like to know that Sign In through Google is not working in Gitpod so how will I check whether the Function I created is correct or not.
Please Help @tachyons, @JoshHeng  and @Ananthavijay

**Comment by Ananthavijay** [comment]
To use the sign in with Google, Facebook, Github option, first you need to create respective apps.
Instead of this run ````bundle exec rake db:seed````. After this run ````bundle exec rails s````.

**Comment by Shivansh2407** [comment]
Although a user's projects can be viewed and edited on their projects page, I have created a link for an Open Online button which takes the user straight to this page. I guess this has been fixed . Please review and merge @Ananthavijay , @tachyons and @satu0king .

**Comment by Shivansh2407** [comment]
Also @Ananthavijay I am unable to view the Comment section in Gitpod when I login. So please let me know how to see the changes that I have committed

**Comment by Ananthavijay** [comment]
There is a problem with the Comment section in CircuitVerse right now.
Refer to this issue: https://github.com/CircuitVerse/CircuitVerse/issues/767

**Comment by Shivansh2407** [comment]
I have fixed the link to open projects online which take user to Circuits page. If user is not logged in then , the user is redirected to login . Please review 👍 @satu0king , @tachyons .

File: app/views/simulator/edit.html.erb [hunk 1]
```diff
@@ -27,6 +27,13 @@
             <li class="toolbarButton logixButton" id="save"><a>Save Online</a></li>
             <li class="toolbarButton logixButton" id="saveOffline"><a>Save Offline</a></li>
             <li class="toolbarButton logixButton" id="createOpenLocalPrompt"><a>Open Offline</a></li>
+            <li class="toolbarButton logixButton" id="createOpenOnlinePrompt">
+              <% if user_signed_in? %>
+                  <a href="<%= user_path(current_user) %>">Open Online</a>
```
Comment (tachyons) [inline app/views/simulator/edit.html.erb]: Do we really need this as we aare already redirecting to login path when not logged in

Comment (Shivansh2407) [reply app/views/simulator/edit.html.erb]: Actually it is redirecting the user to open its own projects. Hence when the user is not logged in ,  then the user is re-directed to login . Please let me know your views on it.

Comment (tachyons) [reply app/views/simulator/edit.html.erb]: That means behavior is still the same except we will lose the page to which we need to redirect once the user is logged in

Comment (Shivansh2407) [reply app/views/simulator/edit.html.erb]: So @tachyons Sir what should be done here?

Comment (tachyons) [reply app/views/simulator/edit.html.erb]: Just Don't show this menu when the user is not logged in

Comment (Shivansh2407) [reply app/views/simulator/edit.html.erb]: Ok @tachyons Sir. Will fix this Soon.

**Comment by Shivansh2407** [comment]
@tachyons Sir Done . Please Review 👍

**Comment by Shivansh2407** [comment]
@nitin10s and @tachyons Please review :+1:

File: app/views/simulator/edit.html.erb (hunk 1 above)
Comment (tachyons) [reply app/views/simulator/edit.html.erb]: You may need to add condition befpre li block to avoid having an empty menu item

**Comment by Shivansh2407** [comment]
Done @tachyons Sir 👍

**Comment by Shivansh2407** [comment]
@tachyons it is showing some merge conflict . What should be done here?

**Comment by aayush-05** [comment]
The edit.html file might got updated after you have made this pr. Either try with the online web editor github provides to fix merge conflicts, or try pulling from master, fixing merge conflicts locally and commiting again.

**Comment by Shivansh2407** [comment]
> The edit.html file might got updated after you have made this pr. Either try with the online web editor github provides to fix merge conflicts, or try pulling from master, fixing merge conflicts locally and commiting again.

Sure 👍

File: app/views/simulator/edit.html.erb [hunk 2]
```diff
@@ -4,295 +4,302 @@
...
+        <a href="#" class="dropdown-toggle text-light" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Project<span class="caret"></span></a>
+        <ul class="dropdown-menu">
+          <li><a class="dropdown-item" id="newProject">New Project</a></li>
+          <li><a class="dropdown-item" id="clearProject">Clear Project</a></li>
+          <li><a class="dropdown-item" id="save">Save Online</a></li>
+          <li><a class="dropdown-item" id="saveOffline">Save Offline</a></li>
+          <li><a class="dropdown-item" id="createOpenLocalPrompt">Open Offline</a></li>
+          <% if user_signed_in? %>
+          <li class="toolbarButton logixButton" id="createOpenOnlinePrompt">
+                <a href="<%= user_path(current_user) %>">Open Online</a>
```
Comment (satu0king) [inline app/views/simulator/edit.html.erb]: @Shivansh2407 actually, this is not the `open online` that was intended. Basically, we need a way to access the project page of the project. Rather it should be `view project page`.  see `project_path`

Comment (satu0king) [reply app/views/simulator/edit.html.erb]: Also why are there so many changes in this PR, ideally you just be adding a link to the drop down.

Comment (Shivansh2407) [reply app/views/simulator/edit.html.erb]: Actually I have added link and did some changes too in data.js file but afterwards only change remaining was in the edit.html.erb file. I will be fixing this issue soon .

**Comment by satu0king** [comment]
I will close this PR, please check original issue for update

**Comment by Shivansh2407** [comment]
Ok @satu0king

**Pull request closed without merging**